src/compiler_lectures/how_salsa_works.md

```diff
@@ -0,0 +1,169 @@
+# How Salsa works
+
+This chapter is based on the explanation given by Niko Matsakis in this [video](https://www.youtube.com/watch?v=_muY4HjSqVw) about [Salsa](https://github.com/salsa-rs/salsa).
+
+## What is Salsa?
+
+Salsa is a library for incremental recomputation, this means reusing computation that has already been done in the past to increase the efficiency of future computations.
```

Comment:
Perhaps add a note that it is not used directly in rustc, but it is used extensively for rust-analyzer and may be integrated into the compiler in the future

Reply:
Exactly what @mark-i-m is mentioning here is what I was going to ask :).